commands/compile/compile.go

```diff
@@ -315,3 +318,15 @@ func maybePurgeBuildCache() {
 	buildcache.New(paths.TempDir().Join("arduino", "cores")).Purge(cacheTTL)
 	buildcache.New(paths.TempDir().Join("arduino", "sketches")).Purge(cacheTTL)
 }
+
+// removeBuildFromSketchFiles removes the files contained in the build directory from
+// the list of the sketch files
+func removeBuildFromSketchFiles(files paths.PathList, build *paths.Path) paths.PathList {
+	var res paths.PathList
+	for _, file := range files {
+		if in, err := file.IsInsideDir(build); !in && err == nil {
```

Comment:
Could we make sure users are informed of this exclusion? I'm thinking just about adding some log

Reply:
Thanks for the suggestion!